### PR TITLE
Don't attempt to capture `self` when all we want to know is the origi…

### PR DESCRIPTION
**Pull request by MikeNachbaur**
…nating class/selector

### CODE REVIEW

**Comment by sgoldberg-sfdc** [comment]
Looks great to me.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/Logging/SFLogger.m [hunk 1]
```diff
@@ -264,7 +264,7 @@ - (void)log:(SFLogLevel)level format:(NSString *)format, ... {
                 file:nil
             function:nil
                 line:0
-                 tag:[[SFLogTag alloc] initWithClass:self.class sender:self]
+                 tag:[[SFLogTag alloc] initWithClass:self.class selector:nil]
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/Logging/SFLogger.m]: Should this be ```nil```?

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/Logging/SFLogger.m]: Never mind. Got it.

**Comment by bhariharan** [comment]
👍

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/Logging/SFLogger.m (hunk 1 above)
Comment (MikeNachbaur) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/Logging/SFLogger.m]: Yes; we don't know what the selector is within this context.  This is to support the legacy NSObject category methods (which are inherently slower and provide less information than by using the macros).

Essentially we need to know "What's the class originating the log message, and what function / method is calling it?"  In this method here, we don't know that without analyzing the backtrace of the call.  Instead, by using the macros, it's able to glean that additional information properly.

**Pull request merged**